pose_prediction/plugin.cpp

```diff
@@ -31,29 +31,29 @@ class pose_prediction_impl : public pose_prediction {
 	}
 
     virtual pose_type get_true_pose() const override {
-		const pose_type* pose_ptr = _m_true_pose->get_latest_ro();
+		switchboard::ptr<const pose_type> pose_ptr = _m_true_pose.get_nullable();
 		return correct_pose(
 			pose_ptr ? *pose_ptr : pose_type{
-				.sensor_time = std::chrono::system_clock::now(),
-				.position = Eigen::Vector3f{0, 0, 0},
-				.orientation = Eigen::Quaternionf{1, 0, 0, 0},
+				std::chrono::system_clock::now(),
+				Eigen::Vector3f{0, 0, 0},
+				Eigen::Quaternionf{1, 0, 0, 0},
```

Comment:
I quite liked the previous format where the members were named, as it was very readable. Any particular reason for the change?

Reply:
This change is necessary because we moved from structs -> classes because data passed over switchboard needs to inherit `switchboard::event`. This allows us to implement [type erasure](https://en.wikipedia.org/wiki/Type_erasure) for switchboard topics without [type-punning](https://en.wikipedia.org/wiki/Type_punning) (which is undefined behavior) and call the correct destructor.

Note that naming the members in classes (like in structs) is possible in C++20 using [designated initializers](https://mariusbancila.ro/blog/2020/02/27/c20-designated-initializers/).